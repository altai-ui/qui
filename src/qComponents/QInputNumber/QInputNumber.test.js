import Component from './src/QInputNumber';

describe('QInputNumber', () => {
  it('should match snapshot', () => {
    const { element } = shallowMount(Component);

    expect(element).toMatchSnapshot();
  });

  it('should match snapshot without controls', () => {
    const { element } = shallowMount(Component, {
      propsData: {
        controls: false
      }
    });

    expect(element).toMatchSnapshot();
  });

  describe('computed', () => {
    it('increaseClass', async () => {
      const instance = shallowMount(Component, {
        propsData: {
          value: 9007199254740992
        }
      });

      expect(instance.vm.increaseClass).toEqual(
        'q-input-number__button_is-disabled'
      );
    });

    it('decreaseClass', async () => {
      const instance = shallowMount(Component, {
        propsData: {
          value: -9007199254740992
        }
      });

      expect(instance.vm.decreaseClass).toEqual(
        'q-input-number__button_is-disabled'
      );
    });
  });

  describe('methods', () => {
    describe('handleChangeNumberButtonClick', () => {
      it('increase', () => {
        const instance = shallowMount(Component, {
          propsData: {
            value: 0
          }
        });
        instance.vm.handleChangeNumberButtonClick(true);

        expect(instance.emitted().change[0]).toEqual([1]);
      });

      it('decrease', () => {
        const instance = shallowMount(Component, {
          propsData: {
            value: 0
          }
        });
        instance.vm.handleChangeNumberButtonClick();

        expect(instance.emitted().change[0]).toEqual([-1]);
      });
    });

    describe('getLocaleSeparator', () => {
      it('should return decimal', () => {
        const { vm } = shallowMount(Component);

        expect(vm.getLocaleSeparator('decimal')).toEqual('.');
      });

      it('should return thousands', () => {
        const { vm } = shallowMount(Component);

        expect(vm.getLocaleSeparator()).toEqual(',');
      });
    });

    it('parseLocaleNumber', () => {
      const { vm } = shallowMount(Component);

      expect(vm.parseLocaleNumber('123,123.0')).toEqual(123123);
    });

    describe('getLocaleSeparator', () => {
      it('should return 2 parts', () => {
        const { vm } = shallowMount(Component);

        expect(vm.getFormattedPartsLength('1,120,120', 5)).toEqual(2);
      });

      it('should return 1 part', () => {
        const { vm } = shallowMount(Component);

        expect(vm.getFormattedPartsLength('1,234', 1)).toEqual(1);
      });
    });

    describe('checkStringAdditions', () => {
      it('should return prefix', () => {
        const { vm } = shallowMount(Component, {
          propsData: {
            prefix: '>'
          }
        });

        expect(vm.checkStringAdditions('>1120<', 'prefix')).toBeTruthy();
      });

      it('should return suffix', () => {
        const { vm } = shallowMount(Component, {
          propsData: {
            suffix: '<'
          }
        });

        expect(vm.checkStringAdditions('>1120<', 'suffix')).toBeTruthy();
      });
    });

    it('getValueWithoutAdditions', () => {
      const { vm } = shallowMount(Component, {
        propsData: {
          prefix: '>',
          suffix: '<'
        }
      });

      expect(vm.getValueWithoutAdditions('>1120<')).toEqual('1120');
    });

    describe('isCharReadonly', () => {
      it('should return false if char is number', () => {
        const { vm } = shallowMount(Component);

        expect(vm.isCharReadonly('1')).toBeFalsy();
      });

      it('should return true if char is -', () => {
        const { vm } = shallowMount(Component);

        expect(vm.isCharReadonly('.')).toBeTruthy();
      });
    });
  });
});
