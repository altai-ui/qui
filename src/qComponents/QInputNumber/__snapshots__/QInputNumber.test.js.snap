// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QInputNumber should match snapshot 1`] = `
<div
  class="q-input-number q-input-number_with-controls"
>
  <button
    class="q-input-number__button_decrease q-input-number__button q-icon-minus"
  />
   
  <q-input-stub
    autocomplete="off"
    class="q-input-number__input"
    label=""
    suffixicon=""
    tabindex=""
    type="text"
    value=""
  />
   
  <button
    class="q-input-number__button_increase q-input-number__button q-icon-plus"
  />
</div>
`;

exports[`QInputNumber should match snapshot without controls 1`] = `
<div
  class="q-input-number"
>
  <!---->
   
  <q-input-stub
    autocomplete="off"
    class="q-input-number__input"
    label=""
    suffixicon=""
    tabindex=""
    type="text"
    value=""
  />
   
  <!---->
</div>
`;
